Patch-Source: https://github.com/calf-studio-gear/calf/commit/76b11e6e1f616034071fcf5f1ed45cb82c3979a3
--
From b7e423628f32db3e6a9ea9bb2d81071d44773a80 Mon Sep 17 00:00:00 2001
From: Violet Purcell <vimproved@inventati.org>
Date: Thu, 21 Sep 2023 18:16:35 -0400
Subject: [PATCH] Replace use of std::bind2nd with std::bind

std::bind2nd was deprecated in C++11 and removed in C++17. Remove usage
of it and replace with std::bind.

Signed-off-by: Violet Purcell <vimproved@inventati.org>
---
 src/calf/orfanidis_eq.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/calf/orfanidis_eq.h b/src/calf/orfanidis_eq.h
index 47abf5bea..61e51ec76 100644
--- a/src/calf/orfanidis_eq.h
+++ b/src/calf/orfanidis_eq.h
@@ -748,7 +748,7 @@ class EllipticTypeBPFilter : public BPFilter {
 			std::vector<eq_double_t> v = landen(k, tol);
 
 			std::transform(v.begin(), v.end(), v.begin(),
-			    bind2nd(std::plus<eq_double_t>(), 1.0));
+			    bind(std::plus<eq_double_t>(), std::placeholders::_1, 1.0));
 
 			K = std::accumulate(v.begin(), v.end(),
 			    1, std::multiplies<eq_double_t>()) * M_PI/2.0;
@@ -764,7 +764,7 @@ class EllipticTypeBPFilter : public BPFilter {
 			std::vector<eq_double_t> vp = landen(kp, tol);
 
 			std::transform(vp.begin(), vp.end(), vp.begin(),
-			    bind2nd(std::plus<eq_double_t>(), 1.0));
+			    bind(std::plus<eq_double_t>(), std::placeholders::_1, 1.0));
 
 			Kprime = std::accumulate(vp.begin(), vp.end(),
 			    1.0, std::multiplies<eq_double_t>()) * M_PI/2.0;
